src/test/java/org/broadinstitute/hellbender/tools/funcotator/FuncotatorDataSourceDownloaderIntegrationTest.java

```diff
@@ -0,0 +1,134 @@
+                        true,
+                        false,
+                        false
+                },
+                {
+                        FuncotatorDataSourceDownloader.SOMATIC_ARG_LONG_NAME,
+                        true,
+                        false,
+                        true
+                }
+        };
+    }
+
+    //==================================================================================================================
+    // Tests:
+
+    @Test(enabled = doDebugTests,
+            dataProvider = "provideForTestDownload",
+            groups = {"funcotatorValidation", "bucket"}
+    )
+    void testDownloadRealDataSources(final String dsTypeArg, final boolean doOverwrite, final boolean doValidate, final boolean doExtract) {
+        final ArgumentsBuilder arguments = new ArgumentsBuilder();
+
+        arguments.addBooleanArgument(dsTypeArg, true);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.OVERWRITE_ARG_LONG_NAME, doOverwrite);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.VALIDATE_INTEGRITY_ARG_LONG_NAME, doValidate);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.EXTRACT_AFTER_DOWNLOAD, doExtract);
+        arguments.addArgument("verbosity", "INFO");
+
+        runCommandLine(arguments);
+    }
+
+    @Test(dataProvider = "provideForTestDownloadSmallDummyDataSources",
+            groups = {"bucket"})
+    void testDownloadDummySmallDataSources(final boolean doOverwrite, final boolean doValidate, final boolean doExtract) {
+
+        // Create an output location for the data sources to go:
+        final File tmpDir = createTempDir("FuncotatorDataSourceDownloaderIntegrationTest_testDownloadDummySmallDataSources");
+        tmpDir.deleteOnExit();
+        final Path tmpDirPath = tmpDir.toPath();
+
+        final Path outputDataSourcesPath = tmpDirPath.resolve(IOUtils.getPath(FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ).getFileName());
+
+        final ArgumentsBuilder arguments = new ArgumentsBuilder();
+
+        arguments.addArgument(FuncotatorDataSourceDownloader.TESTING_OVERRIDE_PATH_FOR_DATA_SOURCES_ARG, FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ);
+        arguments.addArgument(FuncotatorDataSourceDownloader.TESTING_OVERRIDE_PATH_FOR_DATA_SOURCES_SHA256_ARG, FuncotatorTestConstants.DUMMY_DATA_SOURCES_TAR_GZ_SHA256_FILE);
+
+        arguments.addArgument(StandardArgumentDefinitions.OUTPUT_SHORT_NAME, outputDataSourcesPath.toFile().getAbsolutePath());
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.OVERWRITE_ARG_LONG_NAME, doOverwrite);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.VALIDATE_INTEGRITY_ARG_LONG_NAME, doValidate);
+        arguments.addBooleanArgument(FuncotatorDataSourceDownloader.EXTRACT_AFTER_DOWNLOAD, doExtract);
+        arguments.addArgument("verbosity", "INFO");
+
+        runCommandLine(arguments);
```

Comment:
Shouldn't you assert something after running the command line (like the destination existing, etc.)?

Reply:
Sure.  I can't assert that the contents are correct because we shouldn't check in the data sources to git lfs, but I'll add in an existence check for the downloaded file.

Fixed.